### DISCUSSION TITLE
Populate multiple entries in nested Prisma db seed And new Date().toLocateString ?

### DISCUSSION THREAD

**Discussion by AeroFiTech** [prisma/prisma#11723] [Q&A]
Hi guys, I am just trying our Prisma seed - As I am tired of populating the data manually.

I have three model as such

``` 
model User {
  id        Int      @id @default(autoincrement())
  email     String   @db.VarChar(128)
  role      String   @db.VarChar(32)
  title     String?  @db.VarChar(16)
  firstName String   @db.VarChar(128)
  lastName  String   @db.VarChar(128)
  status    String   @db.VarChar(16)
  createdAt DateTime @default(now())
  units     Unit[]
  bookings BookingTrx[]
  @@index([id])
}

model Unit {
  id         Int    @id @default(autoincrement())
  unitNo     String @db.VarChar(16) @unique
  unitId     Int 
  users       User[]   @relation(fields: [unitId], references: [id])
  bookings BookingTrx[]
  @@index([id])
  //  , onDelete: Cascade
}

model BookingTrx {
  id Int @id @default(autoincrement())
  bookedDate DateTime @db.Date
  status  String @db.VarChar(16)
  createdAt DateTime @default(now())
  unit  Unit @relation(fields: [BookingTrxId], references: [id])
  user User @relation(fields: [BookingUserTrxId], references: [id])
  BookingTrxId Int
  BookingUserTrxId Int
  @@index([id])
}
```

This is my db seed algorithm
``` 
const unitData = [...Array(BLOCKS)].map(async (_, i) => {
    [...Array(FLOORS)].map(async (_, j) => {
      [...Array(UNITS)].map(async (_, k) => {
        await prisma.user.create({
          data: {
            email: faker.internet.email(),
            role: Math.random() > 0.7 ? "owner" : "tenant",
            title: Math.random() > 0.6 ? "Mr" : "Mrs",
            firstName: faker.name.firstName(),
            lastName: faker.name.lastName(),
            status: Math.random() < 0.9 ? "active" : "suspended",
            createdAt: new Date(),
            units: {
              create: [
                {
                  unitNo: `${String.fromCharCode(65 + i)}-${j + 1}-${k + 1}`,
                  unitId: k+1,
                  bookings:{
create: {}}
```

Are there any way that I can create multiple (random number) bookings object, i.e. Math.random()*5 for the bookings in each iteration of the units creation?
Besides, how can we force the createdAt @default(now()) function to return date time in local timezone format but not UTC+0 format? I have tried .toLocaleString() and format/parseISO with date-fns library with no avail.

Thanks peeps!

**Comment by pantharshit00** [comment]
Hello @AeroFiTech, our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

> Are there any way that I can create multiple (random number) bookings object, i.e. Math.random()*5 for the bookings in each iteration of the units creation?

You will need to create this inside of the second map. There is no way of indicating to Prisma to create random data seed.

> Besides, how can we force the createdAt @default(now()) function to return date time in local timezone format but not UTC+0 format? I have tried .toLocaleString() and format/parseISO with date-fns library with no avail.

Please try adding `@db.TimeTz(6)` to createAt to use locale information in database.

**Reply by vladi-stevanovic** [reply]
:wave: Hello @AeroFiTech , I’m going through pending GitHub Discussions to check if you need any help. Is this issue still open?

Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! :raised_hands:

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
